src/Streamly/Streams.hs

```diff
@@ -751,7 +751,7 @@ instance Monad m => Functor (ZipStream m) where
         in m Nothing stp yield
 
 instance Monad m => Applicative (ZipStream m) where
-    pure a = ZipStream $ scons a Nothing
+    pure a = ZipStream $ let x = scons a (Just x) in x
```

Comment:
it may be a good idea to factor out this pattern in a function called `srepeat` that will construct a stream by repeating a pure element just like `scons` constructs a singleton stream, and use `srepeat` at both these places. You can put that function next to `scons` definition.